zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/LearnerHandler.java

```diff
@@ -192,6 +194,9 @@ public boolean equals(Object o) {
     private final BufferedInputStream bufferedInput;
     private BufferedOutputStream bufferedOutput;
 
+    private static final int BUFFERED_MESSAGE_SIZE = 10;
```

Comment:
Did you consider to make this tunable?
I am fine to have it NOT tunable.
How did you choose 10 as magic number?

Reply:
The current default value is somehow arbitrary chosen and works for us but I agree this should be tunable. Will make the change.

Reply:
Added a system property to control this setting, as well as documentation.